File: java/src/main/java/vsdk/toolkit/raycasting/photonMap/PhotonFlags.java
/**
Photon flags used by the photon map
*/

package vsdk.toolkit.raycasting.photonMap;

public final class PhotonFlags {
    public static final short DIRECT_LIGHT_PHOTON = 0x10;
    public static final short CAUSTIC_LIGHT_PHOTON = 0x20; // Lower 4 bits reserved for kd tree
    // This type of photon should not be included in the importance sampling
    public static final short NO_IMPSAMP_PHOTON = (short)(DIRECT_LIGHT_PHOTON | CAUSTIC_LIGHT_PHOTON);

    private PhotonFlags() {
    }
}

File: typescript/src/main/vsdk/toolkit/raycasting/photonMap/PhotonFlags.ts
/**
Photon flags used by the photon map
*/
export class PhotonFlags {
  public static readonly DIRECT_LIGHT_PHOTON = 0x10;
  public static readonly CAUSTIC_LIGHT_PHOTON = 0x20;
  public static readonly NO_IMPSAMP_PHOTON =
    PhotonFlags.DIRECT_LIGHT_PHOTON | PhotonFlags.CAUSTIC_LIGHT_PHOTON;

  private constructor() {
  }
}

